File: c/klc_cocoa.c
#include "klc_cocoa.h"
/* dummy c file for platforms that don't have cocoa */

KLCNCOCOAInterface* KLCN_initCOCOA() {
  return NULL;
}

void KLC_deleteCOCOAInterface(KLC_header*, KLC_header**) {
}

File: c/klc_cocoa.m
#include "klc_cocoa.h"
#import <Foundation/Foundation.h>


@interface KLC_ApplicationDelegate : NSObject<NSApplicationDelegate, NSWindowDelegate> {
  NSWindow * window;
}
@end

@implementation KLC_ApplicationDelegate : NSObject
- (id)init {
  if (self = [super init]) {
  }
  return self
}
@end


extern KLC_typeinfo KLC_typeCOCOAInterface;

KLCNCOCOAInterface* KLCN_initCOCOA() {
  KLCNCOCOAInterface* cocoa =
    (KLCNCOCOAInterface*) malloc(sizeof(KLCNCOCOAInterface));
  KLC_init_header(&cocoa->header, &KLC_typeCOCOAInterface);
  return cocoa;
}

void KLC_deleteCOCOAInterface(KLC_header* robj, KLC_header** dq) {
}
